myaws/ecs_node_renew.go

```diff
@@ -118,6 +119,42 @@ func (client *Client) ECSNodeRenew(options ECSNodeRenewOptions) error {
 		return err
 	}
 
+	// Get a list of instance IDs before auto scaling
+	var oldInstanceIds []*string
+	for _, oldNode := range oldNodes {
+		oldInstanceIds = append(oldInstanceIds, oldNode.Ec2InstanceId)
+	}
+
+	// Get a list of instances after auto scaling
+	allNodes, err := client.findECSNodes(options.Cluster)
+	if err != nil {
+		return err
+	}
+
+	// Get a list of instance IDs after auto scaling
+	var allInstanceIds []*string
+	for _, allNode := range allNodes {
+		allInstanceIds = append(allInstanceIds, allNode.Ec2InstanceId)
+	}
+
+	// Select instances to protect from scale in.
+	// By setting "scale-in protection" to instances created at scale-out,
+	// the intended instances (instances created before scale-in) is only terminated at scale-in process.
+	protectInstanceIds, err := client.selectInstanceToProtectFromScaleIn(oldInstanceIds, allInstanceIds)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintln(client.stdout, "Setting scale in protection: ", awsutil.Prettify(protectInstanceIds))
+	// set "scale in protection" to instances created at scale-out.
```

Comment:
Could you add more context to comments why we need to set scale-in protection?
I know it's related to the autoscaling termination policy, but others might not and feels unclear.

Reply:
added comment, thank you.